metadata-transformer/src/main/java/no/unit/nva/metadata/MetadataConverter.java

```diff
@@ -1,67 +1,82 @@
-    private final String jsonld;
+    private final EntityDescription entityDescription;
+    private final int emptyDescriptionHash;
 
-    public MetadataConverter(Model metadata, String jsonld) {
+    public MetadataConverter(Model metadata) {
         this.metadata = metadata;
-        this.jsonld = jsonld;
+        this.entityDescription = new EntityDescription();
+        this.emptyDescriptionHash = entityDescription.hashCode();
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
+    public Optional<CreatePublicationRequest> extractPublicationRequest() {
+        if (metadata.isEmpty()) {
+            return Optional.empty();
+        }
+        prepareDataForTransformation();
+        MetadataExtractor extractor = configureExtractor();
+        for (Statement statement : metadata) {
+            extractor.extract(statement);
+        }
+
+        return entityDescriptionIsPopulated()
+                ? Optional.of(getCreatePublicationRequest())
+                : Optional.empty();
     }
 
-    public CreatePublicationRequest toRequest() throws JsonProcessingException {
-        CreatePublicationRequest request = objectMapper.readValue(jsonld, CreatePublicationRequest.class);
-        getLanguageFromMetadata(metadata).ifPresent(language -> request.getEntityDescription().setLanguage(language));
-        return request;
+    private boolean entityDescriptionIsPopulated() {
+        return emptyDescriptionHash != entityDescription.hashCode();
     }
 
-    private Optional<URI> getLanguageFromMetadata(Model metadata) {
-        return metadata
-            .stream()
-            .filter(this::isDcLanguage)
-            .map(Statement::getObject)
-            .map(Value::stringValue)
-            .map(this::toLexvoUri)
-            .findFirst();
+    private CreatePublicationRequest getCreatePublicationRequest() {
+        CreatePublicationRequest createPublicationRequest = new CreatePublicationRequest();
+        createPublicationRequest.setEntityDescription(entityDescription);
+        return createPublicationRequest;
     }
 
-    private boolean isDcLanguage(Statement statement) {
-        return valueFactory.createIRI(DCTERMS, LANGUAGE).equals(statement.getPredicate());
+    private void prepareDataForTransformation() {
+        metadata.removeIf(statement -> FilterShorterTitles.apply(metadata, statement));
+        metadata.removeIf(statement -> FilterDuplicateContributors.apply(metadata, statement));
     }
 
-    private URI toLexvoUri(String language) {
-        String iso3LanguageCode = ISO3_LANGUAGE_CODE_UNDEFINED;
-        if (!StringUtils.isEmpty(language)) {
-            try {
-                iso3LanguageCode = new Locale(language).getISO3Language();
-            } catch (MissingResourceException e) {
-                logger.warn("Could not map two-letter BCP-47 language code to three-letter ISO639-3 language code.", e);
-            }
-        }
+    private MetadataExtractor configureExtractor() {
```

Comment:
Maybe call this getConfiguredMetadataExtractor, getMetadataExtractor or getDefaultMetadataExtractor to be consistent?

Reply:
I think there is a consistent naming problem, I will rethink this. In general, I like to retain "get/set" for getting/setting fields, though above, I have "getCreatePublicationRequest" because I guess I thought "createCreatePublicationRequest" sounded ridiculous.

Reply:
I would also prefer something without get/set, but I care more that we're consistent with whatever we land on.

Reply:
What about calling them NewPublicationRequest and then have createNewPublicationRequest()

the ambiguity about whether you're creating new requests, publications or publicationrequests remains, but at least it flows of the tongue ;)